src/main.ts

```diff
@@ -55,7 +56,24 @@ export async function run(argv = process.argv.slice(2), options?: Interfaces.Loa
   const cmd = config.findCommand(id)
   if (!cmd) {
     const topic = config.findTopic(id)
-    if (topic) return config.runCommand('help', [id])
+    if (topic) {
+      const subCommands = config.commands.filter(c => (c.id).startsWith(id))
+      const argsOfSubCommands = subCommands.reduce((x, y) => x.concat(y.args), [] as Arg[])
```

Comment:
What's the purpose of this reduce? Is it not possible to just cast subCommands to Arg[]?

Not sure of the right TS syntax but something like `const argsOfSubCommands = subCommands as Array<Arg>` or `const argsOfSubCommands = <Array<Arg>>subCommands`?

Reply:
@RodEsp The point is to get a flat list of all args across all the sub commands.

`subCommands` essentially looks like this:

```
[
  { args: [arg1, arg2] },
  {args: [arg3, arg4] }
]
```

And I want this: `[arg1, arg2, arg3, arg4]` which is why I used `reduce`


Regarding the `as Arg[]`: that was required to get the compiler to be okay with accessing the `arg` property within the reducer

Reply:
Ya, that `as Arg[]` is very standard. 

The reduce makes sense to flatten, but you are just checking the length? Is the question, "is there one matching command with arguments?" Can't you just say `!!subCommands.find(cmd => cmd.args.length > 0)`